_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,338 @@
+
+```java
+public interface TagSearchRepository extends ReactiveElasticsearchRepository<Tag, String>, TagSearchRepositoryInternal {}
+
+interface TagSearchRepositoryInternal {
+    Flux<Tag> search(String query, Pageable pageable);
+}
+
+class TagSearchRepositoryInternalImpl implements TagSearchRepositoryInternal {
+
+    private final ReactiveElasticsearchTemplate reactiveElasticsearchTemplate;
+
+    TagSearchRepositoryInternalImpl(ReactiveElasticsearchTemplate reactiveElasticsearchTemplate) {
+        this.reactiveElasticsearchTemplate = reactiveElasticsearchTemplate;
+    }
+
+    @Override
+    public Flux<Tag> search(String query, Pageable pageable) {
+        List<FieldSortBuilder> builders = new SortToFieldSortBuilderConverter().convert(pageable.getSort());
+
+        NativeSearchQueryBuilder queryBuilder = new NativeSearchQueryBuilder()
+            .withQuery(queryStringQuery(query))
+            .withPageable(PageRequest.of(pageable.getPageNumber(), pageable.getPageSize()));
+
+        builders
+            .stream()
+            .forEach(builder -> {
+                queryBuilder.withSort(builder);
+            });
+
+        NativeSearchQuery nativeSearchQuery = queryBuilder.build();
+        return reactiveElasticsearchTemplate.search(nativeSearchQuery, Tag.class).map(SearchHit::getContent);
+    }
+}
+```
+
+## Learn more about Elastic and JHipster
+
+JHipster helps to simplify the setup of Spring Boot applications or microservices with search capabilities. I hope you enjoyed this quick introduction to Elasticsearh integration options, and could taste the advantages of the Spring Data repository abstraction for encapsulating some basic Elasticsearch operations. Remember to check the [compatiblity matrix](https://docs.spring.io/spring-data/elasticsearch/docs/current/reference/html/#preface.versions) to match the right Spring Data Elasticsearch dependency for your Elasticsearch version. Keep learning, and for more examples and recipes on Elasticsearch and Okta integrations for Spring Boot, check out the following links:
+
+- [Full Stack Java with React, Spring Boot, and JHipster](/blog/2021/11/22/full-stack-java)
+- [Kubernetes to the Cloud with Spring Boot and JHipster](/blog/2021/06/01/kubernetes-spring-boot-jhipster)
+- [Spring Boot and Okta in 2 Minutes](/blog/2020/11/24/spring-boot-okta)
+- [Reactive Java Microservices with Spring Boot and JHipster](/blog/2021/01/20/reactive-java-microservices)
+- [Elasticsearch Clients](https://www.elastic.co/guide/en/elasticsearch/client/index.html)
+- [Get Started with the ELK Stack](/blog/2019/09/26/get-started-elk-stack)
+
+You can find the Blog application code of this tutorial in [GitHub](https://github.com/oktadev/okta-spring-data-elasticsearch-example.).
+
+Be sure to follow us on [Twitter](https://twitter.com/oktadev) and subscribe to our [YouTube Channel](https://youtube.com/c/oktadev) so that you never miss any excellent content!
```

Comment:
```suggestion
Be sure to follow us on [Twitter](https://twitter.com/oktadev) and subscribe to our [YouTube Channel](https://youtube.com/c/oktadev) so that you never miss any of our excellent content!
```